src/server/tiering/disk_storage.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <string>
+#include <system_error>
+
+#include "core/external_alloc.h"
+#include "io/io.h"
+#include "server/io_mgr.h"
+
+namespace dfly::tiering {
+
+struct DiskSegment {
+  size_t offset, length;
+};
```

Comment:
Roman suggested to use `PageRange {offset, page_cnt}` for describing locations. I agree it might make sense, but this is not how all the code is built today:
* ExternalAllocator works with bytes
* *** DiskStorage is here on an abstraction level ***
* High-level code works with bytes

So if we work with pages we would _convert to and from_... which totally defies the point of counting pages?

Reply:
offset, length is fine. please add comments that these are in bytes and should be aligned by page boundaries.